test/jdk/java/awt/Multiscreen/MultiScreenCheckScreenIDTest.java

```diff
@@ -0,0 +1,144 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.AWTException;
+import java.awt.Color;
+import java.awt.GraphicsDevice;
+import java.awt.GraphicsEnvironment;
+import java.awt.Rectangle;
+import java.awt.Robot;
+import java.awt.Window;
+
+import javax.swing.JWindow;
+import javax.swing.SwingUtilities;
+
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.List;
+
+/*
+ * @test
+ * @bug 8280482
+ * @key headful
+ * @summary Test to check if window GC doesn't change within same screen.
+ * @run main MultiScreenCheckScreenIDTest
+ */
+
+public class MultiScreenCheckScreenIDTest extends MouseAdapter {
+    private static final int COLS = 12;
+    private static final int ROWS = 8;
+    private static final Color BACKGROUND = new Color(0, 0, 255, 64);
+    private static GraphicsDevice[] screens;
+    static List<Window> windowList = new ArrayList<>();
+    static Robot robot;
+
+    public static void main(final String[] args) throws Exception {
+        try {
+            createGUI();
+        } finally {
+            for (Window win : windowList) {
+                win.dispose();
+            }
+        }
+        System.out.println("Test Pass");
+    }
+
+    private static void createGUI() throws AWTException {
+        new MultiScreenCheckScreenIDTest().createWindowGrid();
+    }
+
+    private void createWindowGrid() throws AWTException {
+        screens = GraphicsEnvironment
+                .getLocalGraphicsEnvironment()
+                .getScreenDevices();
+
+        if (screens.length < 2) {
+            System.out.println("Testing aborted. Required min of 2 screens. " +
+                    "Available : " + screens.length);
+            return;
+        }
+        robot = new Robot();
+
+        int screenNumber = 1;
+        for (GraphicsDevice screen : screens) {
+            Rectangle screenBounds = screen.getDefaultConfiguration().getBounds();
+
+            for (Rectangle r : gridOfRectangles(screenBounds, COLS, ROWS)) {
+                try {
+                    SwingUtilities.invokeAndWait(() -> {
+                        try {
+                            createWindow(r);
+                        } catch (Exception e) {
+                            throw new RuntimeException(e);
+                        }
+                    });
+                } catch (InterruptedException | InvocationTargetException e) {
+                    e.printStackTrace();
+                }
+                robot.delay(50);
+                robot.waitForIdle();
+                if (windowList.get(windowList.size() - 1).getBounds().intersects
```

Comment:
Maybe it's better to make `createWindow(r)` return a window and cache it instead of several querying of windowList?